ee/hogai/eval/eval_sql.py

```diff
@@ -43,9 +51,73 @@ def _run_eval_sync(self, output, expected=None, **kwargs):
         return Score(name=self._name(), score=score, metadata={"query_generation_retry_count": retry_count})
 
 
-class SQLSyntaxCorrectness(Scorer):
-    """Evaluate if the generated SQL query has correct syntax."""
+class SQLFunctionCorrectness(Scorer):
+    def _name(self):
+        return "sql_function_correctness"
+
+    def _extract_functions_from_sql(self, sql_query: str) -> set[str]:
+        """Extract all function names from SQL query using regex."""
+        if not sql_query:
+            return set()
+
+        # Pattern to match function calls: word followed by opening parenthesis
+        # This captures the function name before the opening parenthesis
+        function_pattern = r"\b([a-zA-Z_][a-zA-Z0-9_]*)\s*\("
 
+        functions = set()
+        for match in re.finditer(function_pattern, sql_query):
+            function_name = match.group(1)
+
+            if function_name.upper() not in SQL_KEYWORDS:
+                functions.add(function_name)
+
+        return functions
+
+    def _is_function_allowed(self, function_name: str) -> bool:
+        return (
+            function_name in HOGQL_CLICKHOUSE_FUNCTIONS
+            or function_name in HOGQL_AGGREGATIONS
+            or function_name in HOGQL_POSTHOG_FUNCTIONS
+        )
```

Comment:
I'd suggest adding something to the `mapping.py` file to do this - if we added a new group of clickhouse functions, we wouldn't know to update these references - we have `_find_function()` with `find_hogql_aggregation`, `find_hogql_function`, and `find_hogql_posthog_function`, for example, we can just expose a `find_any_function(name: str)` in `mapping.py` instead